packages/foundry/src/vault/adapter/aave/aaveV2/AaveV2Adapter.sol

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: GPL-3.0
+// Docgen-SOLC: 0.8.15
+
+pragma solidity ^0.8.15;
+
+import { AdapterBase, IERC20, IERC20Metadata, SafeERC20, ERC20, Math, IStrategy, IAdapter } from "../../abstracts/AdapterBase.sol";
+import { WithRewards, IWithRewards } from "../../abstracts/WithRewards.sol";
+import { ILendingPool, IAaveMining, IAToken } from "./IAaveV2.sol";
+
+/**
+ * @title   AaveV2 Adapter
+ * @author  amatureApe
+ * @notice  ERC4626 wrapper for AaveV2 Vaults.
+ *
+ * An ERC4626 compliant Wrapper for https://github.com/aave/protocol-v2/blob/master/contracts/protocol/lendingpool/LendingPool.sol.
+ * Allows wrapping AaveV2 aTokens with or without an active Liquidity Mining.
+ * Allows for additional strategies to use rewardsToken in case of an active Liquidity Mining.
+ */
+
+contract AaveV2Adapter is AdapterBase, WithRewards {
+  using SafeERC20 for IERC20;
+  using Math for uint256;
+
+  string internal _name;
+  string internal _symbol;
+
+  // @notice The Aave aToken contract
+  IAToken public aToken;
+
+  // @notice The Aave liquidity mining contract
+  IAaveMining public aaveMining;
+
+  // @notice Check to see if Aave liquidity mining is active
+  bool public isActiveMining;
+
+  // @notice The Aave LendingPool contract
+  ILendingPool public lendingPool;
+
+  /*//////////////////////////////////////////////////////////////
+                            CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+  error DifferentAssets(address asset, address underlying);
+
+  /**
+   * @notice Initialize a new AaveV2 Adapter.
+   * @param adapterInitData Encoded data for the base adapter initialization.
+   * @param _aToken Aave wrapped asset. Can also be used to get lendingPool and aaveMining.
+   * @dev `_aToken` - The underlying asset supplied to and wrapped by Aave.
+   * @dev `_lendingPool` - The lending pool.
+   * @dev `_aaveMining` - An optional liquidity mining contract to boost yield.
+   * @dev This function is called by the factory contract when deploying a new vault.
+   */
+
+  function initialize(
+    bytes memory adapterInitData,
+    address,
+    address _aToken
```

Comment:
```suggestion
    bytes memory aaveV2InitData
```

Reply:
This must be in bytes. Since it gets deployed via a some other smart contracts that expect the init params to be (bytes,address,bytes)

Reply:
Yup should've figured